pipeline/authn/authenticator_bearer_token.go

```diff
@@ -107,6 +107,10 @@ func (a *AuthenticatorBearerToken) Authenticate(r *http.Request, session *Authen
 		return helper.ErrForbidden.WithReasonf("The configured extra_from GJSON path returned an error on JSON output: %s", err.Error()).WithDebugf("GJSON path: %s\nBody: %s\nResult: %s", cf.ExtraFrom, body, extraRaw).WithTrace(err)
 	}
 
+	if len(subject) == 0 {
```

Comment:
Before, it would go to `upstream` with an empty subject. Which seems not a desired outcome.

Reply:
Nice!

Reply:
Upon review I'm wondering if we really want to do this? An empty subject could have meaning such as "anonymous" user in some environment. Changing this to error will cause a breaking change for those relying on this feature. What was the intention of changing this?